tests/core/test_emscripten_async_call.c

```diff
@@ -14,7 +14,7 @@ void myatexit() {
 void callback(void *arg) {
   printf("callback\n");
   doneCallback = true;
-  assert((int)arg == 42);
+  assert((long)arg == 42);
```

Comment:
How about instead do that cast on the right: `assert(arg == (void*)42);`?  (or use `inptr_t` on the LHS)